cmd/kube-controller-manager/app/core.go

```diff
@@ -123,7 +123,15 @@ func startNamespaceController(ctx ControllerContext) (bool, error) {
 		return true, fmt.Errorf("failed to parse preferred server resources: %v", err)
 	}
 	discoverResourcesFn := namespaceKubeClient.Discovery().ServerPreferredNamespacedResources
-	if _, found := gvrs[extensions.SchemeGroupVersion.WithResource("thirdpartyresource")]; !found {
+	// Find out whether TPR is enabled despite the version
+	tprEnabled := false
+	for gvr := range gvrs {
+		if gvr.Group == extensions.GroupName && gvr.Resource == "thirdpartyresources" {
```

Comment:
I'm against any special casing for TPRs.  Just make the discovery always dynamic.